src/HttpClient.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Http\Client;
+
+use Psr\Http\Message\StreamInterface;
+use Psr\Http\Message\UriInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Interface for HTTP conventional methods
```

Comment:
then maybe just add a second line saying: "Use this interface when you do not have PSR-7 RequestInterface instances available."